components/brave_shields/browser/ad_block_subscription_service_manager.h

```diff
@@ -0,0 +1,92 @@
+
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "base/macros.h"
+#include "base/memory/scoped_refptr.h"
+#include "base/optional.h"
+#include "base/synchronization/lock.h"
+#include "base/values.h"
+#include "brave/components/brave_component_updater/browser/brave_component.h"
+#include "brave/components/brave_shields/browser/ad_block_subscription_service.h"
+#include "brave/vendor/adblock_rust_ffi/src/wrapper.h"
+#include "third_party/blink/public/mojom/loader/resource_load_info.mojom-shared.h"
+#include "url/gurl.h"
+
+namespace base {
+class ListValue;
+}  // namespace base
+
+class AdBlockServiceTest;
+
+using brave_component_updater::BraveComponent;
+
+namespace brave_shields {
+
+using SubscriptionIdentifier = GURL;
+
+// The AdBlock subscription service manager, in charge of initializing and
+// managing AdBlock clients corresponding to custom filter list subscriptions.
+class AdBlockSubscriptionServiceManager {
+ public:
+  explicit AdBlockSubscriptionServiceManager(BraveComponent::Delegate* delegate);
+  ~AdBlockSubscriptionServiceManager();
+
+  std::vector<FilterListSubscriptionInfo> GetSubscriptions() const;
+  std::string& GetSubscriptionSource(const SubscriptionIdentifier& id);
+  void EnableSubscription(const SubscriptionIdentifier& id, bool enabled);
+  void DeleteSubscription(const SubscriptionIdentifier& id);
+  void RefreshSubscription(const SubscriptionIdentifier& id);
+  void CreateSubscription(const GURL list_url);
+  void RefreshAllSubscriptions();
+
+  bool IsInitialized() const;
+  bool Start();
+  void ShouldStartRequest(const GURL& url,
+                          blink::mojom::ResourceType resource_type,
+                          const std::string& tab_host,
+                          bool* did_match_rule,
+                          bool* did_match_exception,
+                          bool* did_match_important,
+                          std::string* mock_data_url);
+  void EnableTag(const std::string& tag, bool enabled);
+  void AddResources(const std::string& resources);
+  void EnableFilterList(const std::string& uuid, bool enabled);
+
+  base::Optional<base::Value> UrlCosmeticResources(
+          const std::string& url);
+  base::Optional<base::Value> HiddenClassIdSelectors(
+          const std::vector<std::string>& classes,
+          const std::vector<std::string>& ids,
+          const std::vector<std::string>& exceptions);
+
+ private:
+  friend class ::AdBlockServiceTest;
+  bool Init();
+  void StartSubscriptionServices();
+  void UpdateFilterListPrefs(const std::string& uuid, bool enabled);
+
+  brave_component_updater::BraveComponent::Delegate* delegate_;  // NOT OWNED
+  bool initialized_;
+  base::Lock subscriptions_services_lock_;
+  std::map<SubscriptionIdentifier, std::unique_ptr<AdBlockSubscriptionService>> subscription_services_;
+
+  DISALLOW_COPY_AND_ASSIGN(AdBlockSubscriptionServiceManager);
```

Comment:
DISALLOW_COPY_AND_ASSIGN is deprecated.  Use direct
`AdBlockSubscriptionServiceManager(const AdBlockSubscriptionServiceManager&) = delete;`
`AdBlockSubscriptionServiceManager& operator=(const AdBlockSubscriptionServiceManager&) = delete;`